test/e2e/storage/vsphere/vsphere.go

```diff
@@ -0,0 +1,233 @@
+	s := object.NewSearchIndex(vs.Client.Client)
+	vmUUID = strings.ToLower(strings.TrimSpace(vmUUID))
+	//framework.Logf("Unable to find VM by UUID. VM UUID: %s, datacenter: %s, vSphere: %s", vmUUID, dc.Name(), vs.Config.Hostname)
+	return s.FindByUuid(ctx, datacenter, vmUUID, true, nil)
+}
+
+// GetFolderByPath gets the Folder Object Reference from the given folder path
+// folderPath should be the full path to folder
+func (vs *VSphere) GetFolderByPath(ctx context.Context, dc object.Reference, folderPath string) (vmFolderMor types.ManagedObjectReference, err error) {
+	Connect(ctx, vs)
+	datacenter := object.NewDatacenter(vs.Client.Client, dc.Reference())
+	finder := find.NewFinder(datacenter.Client(), true)
+	finder.SetDatacenter(datacenter)
+	vmFolder, err := finder.Folder(ctx, folderPath)
+	if err != nil {
+		glog.Errorf("Failed to get the folder reference for %s. err: %+v", folderPath, err)
+		return vmFolderMor, err
+	}
+	return vmFolder.Reference(), nil
+}
+
+func (vs *VSphere) CreateVolume(volumeOptions *VolumeOptions, dataCenterRef types.ManagedObjectReference) (string, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	Connect(ctx, vs)
+	datacenter := object.NewDatacenter(vs.Client.Client, dataCenterRef)
+	var (
+		err                     error
+		directoryAlreadyPresent = false
+	)
+	if datacenter == nil {
+		err = fmt.Errorf("datacenter is nil")
+		return "", err
+	}
+	if volumeOptions == nil {
+		volumeOptions = &VolumeOptions{}
+	}
+	if volumeOptions.Datastore == "" {
+		volumeOptions.Datastore = vs.Config.DefaultDatastore
+	}
+	if volumeOptions.CapacityKB == 0 {
+		volumeOptions.CapacityKB = DefaultDiskCapacityKB
+	}
+	if volumeOptions.Name == "" {
+		volumeOptions.Name = "e2e-vmdk-" + strconv.FormatInt(time.Now().UnixNano(), 10)
+	}
+	if volumeOptions.DiskFormat == "" {
+		volumeOptions.DiskFormat = DefaultDiskFormat
+	}
+	if volumeOptions.SCSIControllerType == "" {
+		volumeOptions.SCSIControllerType = DefaultSCSIControllerType
+	}
+
+	finder := find.NewFinder(datacenter.Client(), true)
+	finder.SetDatacenter(datacenter)
+	ds, err := finder.Datastore(ctx, volumeOptions.Datastore)
+	if err != nil {
+		err = fmt.Errorf("Failed while searching for datastore: %s. err: %+v", volumeOptions.Datastore, err)
+		return "", err
+	}
+	directoryPath := filepath.Clean(ds.Path(VolDir)) + "/"
+	fileManager := object.NewFileManager(ds.Client())
+	err = fileManager.MakeDirectory(ctx, directoryPath, datacenter, false)
+	if err != nil {
+		if soap.IsSoapFault(err) {
+			soapFault := soap.ToSoapFault(err)
+			if _, ok := soapFault.VimFault().(types.FileAlreadyExists); ok {
+				directoryAlreadyPresent = true
+				glog.V(4).Infof("Directory with the path %+q is already present", directoryPath)
+			}
+		}
+		if !directoryAlreadyPresent {
+			glog.Errorf("Cannot create dir %#v. err %s", directoryPath, err)
+			return "", err
+		}
+	}
+	glog.V(4).Infof("Created dir with path as %+q", directoryPath)
```

Comment:
Same here for the logging, use framework.Logf

Reply:
replaced all `glog` message printing to use `framework.Logf`